hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -482,11 +513,29 @@ extends Importer:
                   case ((pss, ctx), ps) => 
                     val (qs, newCtx) = params(ps)(using ctx)
                     (pss :+ ParamList(ParamListFlags.empty, qs), newCtx)
+              // * Elaborate signature
+              val s = td.signature.orElse(newSignatureTrees.get(id.name)).map(term)
               val b = rhs.map(term(_)(using newCtx))
               val r = FlowSymbol(s"‹result of ${sym}›", nextUid)
-              val tdf = TermDefinition(owner, k, sym, pss,
-                td.signature.orElse(newSignatureTrees.get(id.name)).map(term), b, r)
+              val tdf = TermDefinition(owner, k, sym, pss, s, b, r, TermDefFlags(mod))
               sym.defn = S(tdf)
+
+              // Restrictions regarding functions receiving module parameters
+              
+              // not working because sometimes the signature is N 
```

Comment:
Currently "signatures" are only picked up from externally-given declarations. Picking up result types when they're given requires a bit more logic to be added, probably in `TypeDefImpl`. It will likely be useful to have in general, along with logic that propagates types to arguments from the signature. But that can be done later.